packages/ui/src/models/camel/camel-route-resource.ts

```diff
@@ -46,12 +48,15 @@ export class CamelRouteResource implements CamelResource, BeansAwareResource {
   ) => number;
   private entities: BaseCamelEntity[] = [];
   private resolvedEntities: BaseVisualCamelEntityDefinition | undefined;
-  private comments: string[] = [];
+  private serializer: CamelResourceSerializer;
 
-  constructor(json?: unknown) {
-    if (!json) return;
-    const rawEntities = Array.isArray(json) ? json : [json];
-    this.entities = rawEntities.reduce((acc, rawItem) => {
+  constructor(code?: string, serializer?: CamelResourceSerializer) {
```

Comment:
Can we keep the previous constructor instead? To be consistent with the other Resource?

Reply:
yup , I'm already working on reverting it. Too much changes in tests would be required.